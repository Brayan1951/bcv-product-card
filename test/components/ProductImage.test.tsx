import React from "react";
import renderer from "react-test-renderer";
import { ProductCard } from "../../src/components/ProductCard";
import { ProductImage } from '../../src/components/ProductImage';
import { product2 } from '../data/products';

describe('Product Image', () => {
    test('debe de mostar el componente correctamente con la imagen personalizada', () => {
        const wrapper = renderer.create(<ProductImage img="./no-image.jpg" />)
        expect(wrapper.toJSON()).toMatchSnapshot()

    })
    test('debe de mostrar el componente con el nombre del a imagen del producto', () => {
        const wrapper = renderer.create(<ProductCard initialValues={{}} product={product2}>
            {
                () => (
                    <ProductImage />
                )
            }
        </ProductCard>)
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

})

