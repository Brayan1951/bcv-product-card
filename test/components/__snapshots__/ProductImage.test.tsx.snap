// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image debe de mostar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Coffe"
  className="productImg undefined"
  src="./no-image.jpg"
/>
`;

exports[`Product Image debe de mostrar el componente con el nombre del a imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
